device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -262,23 +262,28 @@ public void open() throws TransportException
     public void close() throws TransportException
     {
         log.debug("Shutting down amqp layer...");
-        closeAsync();
-
         try
         {
-            log.trace("Waiting for reactor to close...");
-            closeReactorLatch.await(MAX_WAIT_TO_CLOSE_CONNECTION, TimeUnit.MILLISECONDS);
+            closeAsync();
```

Comment:
Any code comments worthy to include given there was a bug here earlier? I'm not asserting there is, just asking you to consider. :)